pkg/linksharing/handler.go

```diff
@@ -0,0 +1,156 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package linksharing
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"go.uber.org/zap"
+	"gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/lib/uplink"
+	"storj.io/storj/pkg/ranger"
+	"storj.io/storj/pkg/storj"
+)
+
+var (
+	mon = monkit.Package()
+)
+
+// Handler implements the link sharing HTTP handler
+type Handler struct {
+	log    *zap.Logger
+	uplink *uplink.Uplink
+}
+
+// NewHandler creates a new link sharing HTTP handler
+func NewHandler(log *zap.Logger, uplink *uplink.Uplink) *Handler {
+	return &Handler{
+		log:    log,
+		uplink: uplink,
+	}
+}
+
+// ServeHTTP handles link sharing requests
+func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Comment:
As before the receiver (`h`) should have a longer name

Reply:
:+1: